Epsilon/Services/LearningOutcomeCanvasResultService.cs

```diff
@@ -68,15 +68,18 @@ public async IAsyncEnumerable<LearningDomainSubmission> GetSubmissions(string st
 
         await Task.WhenAll(submissionsTask, domainOutcomesTask);
 
-        foreach (var submission in submissionsTask?.Result?.Courses?.SelectMany(static c => c.Submissions.Nodes))
+        if (submissionsTask.Result?.Courses != null)
```

Comment:
I think this is good, but to reduce nesting maybe you can inverse the if and throw an exception or something like that. It's also better for debugging and error handling